src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -154,13 +154,14 @@ def execute(self, unexpanded_argv):  # pylint: disable=too-many-statements
 
         # Rudimentary parsing to get the command
         nouns = []
-        for noun in argv:
+        for i in range(len(argv)):  # pylint: disable=consider-using-enumerate
             try:
-                if noun[0] == '-':
+                if argv[i][0] == '-':
```

Comment:
This checks each char in the noun?
Would this mean I can't have `-` in my command? (e.g. `az vm my-command`)